src/main/java/nl/igorski/mwengine/MWEngine.java

```diff
@@ -113,6 +121,74 @@ public MWEngine( Context aContext, IObserver aObserver ) {
         MWEngineCore.init(); // registers the interface between this Java class and the native code
     }
 
+    public static int[] cpuCores;
+
+    /**
+     * Call from your Activitity.onCreate()
+     */
+    public static boolean optimizePerformance( Activity activity ) {
+        if ( activity.getWindow() == null ) {
```

Comment:
Create single function that performs several performance improvements (where supported) on application start.